src/notifications/webhookManager.ts

```diff
@@ -0,0 +1,59 @@
+import { SpvWalletClient } from '../client';
+import { Events } from '../types';
+import { EventsMap } from './eventsMap';
+import { EventHandler, RawEvent, WebhookHttpHandler, WebhookOptions } from './interface';
+
+export class WebhookManager {
+  public url: string;
+  public options: WebhookOptions;
+  public subscriber: SpvWalletClient;
+  public handlers: EventsMap;
+
+  constructor(subscriber: SpvWalletClient, url: string, options: WebhookOptions = {}) {
+    this.url = url;
+    this.subscriber = subscriber;
+    this.options = {
+      tokenValue: options.tokenValue || '',
+      tokenHeader: options.tokenHeader || '',
+    };
+    this.handlers = new EventsMap();
```

Comment:
you can pass a logger here (if you want) 🙈 

```suggestion
    this.handlers = new EventsMap(subscriber.logger);
```

Reply:
added